python/GafferImageUI/ImageViewUI.py

```diff
@@ -1063,11 +1107,152 @@ def __stateChanged( self, imageGadget ) :
+
+class _CompareParentPlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plug, **kw ) :
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 0 )
+
+		GafferUI.PlugValueWidget.__init__( self, self.__row, plug, **kw )
+
+		widgets = [ GafferUI.PlugValueWidget.create( p ) for p in plug.children( Gaffer.Plug ) ]
+		for i in widgets[1:]:
+			# Aside from compareMode, the compare widgets don't start visible
+			i.setVisible( False )
+
+			# ViewPlugValueWidget requires us to set gafferAdjoinedLeft on the internal widget.  This
+			# seems to be the case for many widget types, but WidgetAlgo.joinEdges only has a special
+			# case for BoolPlugValueWidget.  Work around this by just setting the property manually
+			i.viewWidget()._qtWidget().setProperty( "gafferAdjoinedLeft", True )
+
+		self.__row[:] = widgets
+
+class _CompareModePlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plug, **kw ) :
+
+		self.__button = GafferUI.MenuButton(
+			image = "compareModeNone.png",
+			menu = GafferUI.Menu(
+				Gaffer.WeakMethod( self.__menuDefinition ),
+				title = "Compare Mode",
+			)
+		)
+		self.__button._qtWidget().setMaximumWidth( 25 )
+
+		GafferUI.PlugValueWidget.__init__( self, self.__button, plug, **kw )
+
+		self.__iconDict = {
+			"" : "compareModeNone.png",
+			"replace" : "compareModeReplace.png",
+			"over" : "compareModeOver.png",
+			"under" : "compareModeUnder.png",
+			"difference" : "compareModeDifference.png",
+			"sideBySide" : "compareModeSideBySide.png"
+		}
+
+		plug.node().viewportGadget().keyPressSignal().connect(
+			Gaffer.WeakMethod( self.__keyPress ),
+			scoped = False
+		)
+
+		self._updateFromPlug()
+
+	def _updateFromPlug( self ) :
+
+		with Gaffer.Context() :
+			v = self.getPlug().getValue()
+
+		if v != "":
+			Gaffer.Metadata.registerValue( self.getPlug(), "imageViewer:lastCompareMode", v )
+
+		icon = self.__iconDict[v] if v in self.__iconDict else "compareModeNone.png"
+		self.__button.setImage( icon )
+
+
+		# We only want to set gafferAdjoinedRight on ourselves when the other controls are visible.
+		# When we're showing by ourselves, we want proper corners on the right edge.
+		# WidgetAlgo.joinEdges doesn't support unjoining edges, so just set the property manually
+		self.__button._qtWidget().setProperty( "gafferAdjoinedRight", v != "" )
+		self._repolish()
+		if self.parent():
+			self.parent()[-1].setVisible( v != "" )
+
+			self.parent()[-1]._repolish()
+
+	def __hotkeyTarget( self ):
+		with Gaffer.Context() :
+			v = self.getPlug().getValue()
+
+		if v == "":
+			return Gaffer.Metadata.value( self.getPlug(), "imageViewer:lastCompareMode" ) or "over"
+		else:
+			return ""
+
+	def __keyPress( self, gadget, event ) :
+
+		if event.key == "D" :
+			self.__setValue( self.__hotkeyTarget() )
+			return True
+
+		return False
+
+	def __menuDefinition( self ) :
+
+		with self.getContext() :
+			compareMode = self.getPlug().getValue()
```

Comment:
Oh, it was originally for that, but I deleted it after realizing that our menus currently don't show both an icon and a checkbox - do you want me to try and fix that?  Do you expect it to be much work?

Reply:
I checked the Qt source (`qstylesheetstyle.cpp` if you're interested) and this looks like a no-go :

```
f (!mi.icon.isNull()) {
   // Draw icon
} else if (mi.menuHasCheckableItems) {
   // Draw checkbox
}
```

Which is a pity, because I really like the icons but I think the checkbox would be more useful to me.

Reply:
![compareWipeUI](https://user-images.githubusercontent.com/27969079/188215690-332822b7-4352-487b-a521-80173dd1b9ab.png)

I was trying to figure out if it would be possible to show which one is selected with an icon, but it's tough when the icon is just16x16 pixels

Reply:
Using the new icons was part of 1331762 - not sure if this is the best approach, but it feels tolerable to me?